crates/ide-assists/src/utils.rs

```diff
@@ -362,20 +362,20 @@ pub(crate) fn find_struct_impl(
     });
 
     if let Some(ref impl_blk) = block {
-        if has_fn(impl_blk, name) {
+        if has_any_fn(impl_blk, names) {
             return None;
         }
     }
 
     Some(block)
 }
 
-fn has_fn(imp: &ast::Impl, rhs_name: &str) -> bool {
+fn has_any_fn(imp: &ast::Impl, names: Vec<String>) -> bool {
     if let Some(il) = imp.assoc_item_list() {
         for item in il.assoc_items() {
             if let ast::AssocItem::Fn(f) = item {
                 if let Some(name) = f.name() {
-                    if name.text().eq_ignore_ascii_case(rhs_name) {
+                    if names.contains(&name.text().to_string().to_ascii_lowercase()) {
```

Comment:
```suggestion
                    if names.iter().any(|n| n.eq_ignore_ascii_case(name.text())) {
```
This is a lot cheaper as we save on the `to_string` and `to_ascii_lowercase` allocations

Reply:
Nice optimization!